File: caesar_cipher_encryptor/caesar_cipher_encryptor.go
package main

import (
	"fmt"
)

func caesarCipherEncryptor(str string, key int) string {
	newLetters := []rune{}
	newKey := key % 26

	for _, letter := range str {
		newLetter := getNewLetter(letter, newKey)
		newLetters = append(newLetters, newLetter)
	}

	return string(newLetters)
}

func getNewLetter(letter rune, key int) rune {
	newCharCode := int(letter) + key
	if newCharCode <= 'z' {
		return rune(newCharCode)
	}
	return rune('a' + (newCharCode - 'z' - 1))
}

func main() {
	fmt.Println(caesarCipherEncryptor("xyz", 2))
}

File: caesar_cipher_encryptor/caesar_cipher_encryptor.java
import java.util.*;

public class CaesarCipher {

    public static String caesarCipherEncryptor(String str, int key) {
        StringBuilder newLetters = new StringBuilder();
        int newKey = key % 26;
        for (char letter : str.toCharArray()) {
            char newLetter = getNewLetter(letter, newKey);
            newLetters.append(newLetter);
        }
        return newLetters.toString();
    }

    private static char getNewLetter(char letter, int key) {
        int newCharCode = letter + key;
        return newCharCode <= 'z' ? (char) newCharCode : (char) ('a' + newCharCode - 'z' - 1);
    }

    public static void main(String[] args) {
        System.out.println(caesarCipherEncryptor("xyz", 2));
    }
}
